backend/src/bundles/auth/auth.service.ts

```diff
@@ -11,10 +20,59 @@ class AuthService {
         this.userService = userService;
     }
 
-    public signUp(
+    public async signUp(
         userRequestDto: UserSignUpRequestDto,
     ): Promise<UserSignUpResponseDto> {
-        return this.userService.create(userRequestDto);
+        const foundUserByEmail = await this.userService.findByEmail(
+            userRequestDto.email,
+        );
+        if (foundUserByEmail) {
+            throw new HttpError({
+                message: ExceptionMessage.EMAIL_TAKEN,
+                status: HttpCode.BAD_REQUEST,
+            });
+        }
+        const { id } = await this.userService.create(userRequestDto);
+
+        const user = await this.userService.getUserWithProfile(id);
+
+        return {
+            accessToken: generateToken({ id }),
+            user,
+        };
+    }
+
+    public async login({
+        email,
+        password,
+    }: UserSignInRequestDto): Promise<UserSignInResponseDto> {
+        const foundUserByEmail = await this.userService.findByEmail(email);
+
+        if (!foundUserByEmail) {
+            throw new HttpError({
+                message: ExceptionMessage.USER_NOT_FOUND,
+                status: HttpCode.BAD_REQUEST,
+            });
+        }
+        const { passwordHash, id } = foundUserByEmail;
+        const isEqualPassword = await comparePasswords(password, passwordHash);
```

Comment:
It looks like this function should accept 2 passwords, but in fact it doesn't.
Are you sure this function works correctly, it doesn't take into account passwordSalt...